server/src/test/java/org/opensearch/tasks/consumer/TopNSearchTasksLoggerTests.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tasks.consumer;
+
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.core.LogEvent;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.opensearch.action.search.SearchShardTask;
+import org.opensearch.common.logging.Loggers;
+import org.opensearch.common.logging.MockAppender;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.tasks.ResourceStats;
+import org.opensearch.tasks.ResourceStatsType;
+import org.opensearch.tasks.ResourceUsageMetric;
+import org.opensearch.tasks.Task;
+import org.opensearch.test.OpenSearchSingleNodeTestCase;
+
+import java.util.Collections;
+
+import static org.opensearch.tasks.consumer.TopNSearchTasksLogger.LOG_TOP_QUERIES_FREQUENCY;
+import static org.opensearch.tasks.consumer.TopNSearchTasksLogger.LOG_TOP_QUERIES_SIZE;
+
+public class TopNSearchTasksLoggerTests extends OpenSearchSingleNodeTestCase {
+    static MockAppender appender;
+    static Logger searchLogger = LogManager.getLogger(TopNSearchTasksLogger.TASK_DETAILS_LOG_PREFIX + ".search");
+
+    private TopNSearchTasksLogger topNSearchTasksLogger;
+
+    @BeforeClass
+    public static void init() throws IllegalAccessException {
+        appender = new MockAppender("trace_appender");
+        appender.start();
+        Loggers.addAppender(searchLogger, appender);
+    }
+
+    @AfterClass
+    public static void cleanup() {
+        appender.stop();
+        Loggers.removeAppender(searchLogger, appender);
+    }
+
+    public void testLoggerWithTasks() {
+        final Settings settings = Settings.builder().put(LOG_TOP_QUERIES_SIZE, 1).put(LOG_TOP_QUERIES_FREQUENCY, 10).build();
+        topNSearchTasksLogger = new TopNSearchTasksLogger(settings);
+
+        generateTasks(5);
+        // sleeping enough time for the logger to complete publishing entries to log file
```

Comment:
This test relies on a frequency setting and is going to be flakey by design. Maybe a frequency setting of 0 could make it synchronous?

Reply:
Updated the test to use 0ms as frequency setting.